### PR TITLE
Improve activemq persistence

### PR DESCRIPTION
**Pull request by Marcel-Jansen**

### CODE REVIEW

File: charts/gxf/Chart.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+version: '1.2.4'
 icon: https://artwork.lfenergy.org/projects/grid-exchange-fabric/abbrev/color/grid-exchange-fabric-abbrev-color.png
 maintainers:
   - name: OSGP
     email: gxf@lists.lfenergy.org
     url: https://www.lfenergy.org/projects/gxf/
 dependencies:
 - name: activemq-artemis
-  version: "0.0.5"
+  version: "0.0.6"
```
Comment (JelleHoffman) [inline charts/gxf/Chart.yaml:12]: Shouldn't this be version 0.0.7 like in the chart.yaml of activeMQ?

File: charts/activemq-artemis/templates/statefulsets.yaml [hunk 2]
```diff
@@ -35,7 +35,7 @@ spec:
       serviceAccountName: {{ template "fullname" . }}
       containers:
       - name: {{ template "fullname" . }}
-        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        image: "{{ .Values.image }}{{- if .Values.imageTag -}}:{{- end -}}{{ .Values.imageTag }}"
```
Comment (JelleHoffman) [inline charts/activemq-artemis/templates/statefulsets.yaml:38]: What if the chart does not contain an imageTag will the `{{ .Values.imageTag }}` does make an incorrect YAML file. Isn't it better to have this within the if statement?

File: charts/gxf/Chart.yaml (hunk 1 above)
Comment (JelleHoffman) [inline charts/gxf/Chart.yaml:12]: Shouldn't this be version 0.0.7 like in the chart.yaml of activeMQ?

Comment (sanderv) [inline charts/gxf/Chart.yaml:12]:
```suggestion
  version: "0.0.7"
```

**Review by JelleHoffman — CHANGES_REQUESTED** [review]

File: charts/gxf/Chart.yaml (hunk 1 above)
Comment (sanderv) [reply charts/gxf/Chart.yaml:12]: because that's the new version in the activemq chart above

Comment (Marcel-Jansen) [reply charts/gxf/Chart.yaml:12]: The build will fail: https://github.com/OSGP/charts/pull/9/commits/998f4a8a69e5bfbf99b1ca62150b36a277e24747
Version 0.0.7 does not exist in the repository yet. So this should be changed in a next iteration.

Comment (Marcel-Jansen) [reply charts/gxf/Chart.yaml:12]: The build will fail: 998f4a8
Version 0.0.7 does not exist in the repository yet. So this should be changed in a next iteration.

File: charts/activemq-artemis/templates/statefulsets.yaml (hunk 2 above)
Comment (Marcel-Jansen) [reply charts/activemq-artemis/templates/statefulsets.yaml:38]: If the imageTag is empty the YAML will still be valid. The semicolon won't be added. Of course a user is then required to add the imageTag to the image variable. 
This gives an user the freedom to no use the imageTag variable. This is required for our SMHE pipeline as this pipeline updates all imageTag values.

**Review by sanderv — APPROVED** [review]

**Review by JelleHoffman — APPROVED** [review]

**Pull request merged**